src/Aptos/MoveTypes.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <Aptos/MoveTypes.h>
+#include <sstream>
+
+namespace TW::Aptos {
+
+Aptos::ModuleId::ModuleId(Address accountAddress, Identifier name) noexcept
+    : mAccountAddress(accountAddress), mName(std::move(name)) {
+}
+
+Data ModuleId::serialize() const noexcept {
```

Comment:
I think we need to be careful about the future usage of smart contract that we want to add.

https://github.com/aptos-labs/aptos-core/blob/ce633bee7a1270232eded5a9d5176c642f91ccf1/types/src/access_path.rs use `access_vector()`

And access vector is:

```rust
pub fn access_vector(&self) -> Vec<u8> {
        let mut key = vec![CODE_TAG];
        key.append(&mut bcs::to_bytes(self).unwrap());
        key
    }
```

Let's keep both of them separated since they belong to different usage, maybe I should rename `serialize` to `accessVector`